src/MessageBox.Avalonia/Extensions/OpenUrl.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace MessageBox.Avalonia.Extensions
+{
+    public static class OpenUrlExtension
+    {
+        private static bool IsValidUrl(string url)
+        {
+            if (string.IsNullOrWhiteSpace(url)) return false;
+            if (!Uri.IsWellFormedUriString(url, UriKind.Absolute)) return false;
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var tmp)) return false;
+            return tmp.Scheme == Uri.UriSchemeHttp || tmp.Scheme == Uri.UriSchemeHttps;
+        }
+        public static void OpenUrl(this string url)
+        {
+            if (IsValidUrl(url))
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    //https://stackoverflow.com/a/2796367/241446
+                    using (Process proc = new Process { StartInfo = { UseShellExecute = true, FileName = url } })
```

Comment:
It seems that it is better to call the factory method here, which, depending on the OS, will return us an instance

Reply:
How do you want to implement this? May you provide an example?

Reply:
```cs
public static Process CreateProcess(string url)
{
    const string net46AndOlder = "4.0.30319.42000";
    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
    {
        return new Process { StartInfo = { UseShellExecute = true, FileName = url } };
    }
    else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
    {
        return new Process { StartInfo = { FileName = "x-www-browser", Arguments = url } };
    }
    else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
    {
        return new Process { StartInfo = { FileName = "open", Arguments = url } };
    }
    else if (net46AndOlder.Equals(Environment.Version.ToString(), StringComparison.InvariantCulture))
    {
        return new Process { StartInfo = { UseShellExecute = true, FileName = url } };    
    }

    throw new NotSupportedException(
        "Opening URL is only available on the following operating systems: Windows, Linux, OSX");
}
```